src/glibc/sysdeps/unix/sysv/linux/ioctl.c

```diff
@@ -31,7 +32,7 @@ __ioctl (int fd, unsigned long int request, ...)
   void *arg = va_arg (args, void *);
   va_end (args);
 
-        return MAKE_SYSCALL(IOCTL_SYSCALL, "syscall|ioctl", (uint64_t) fd, (uint64_t) request, (uint64_t) arg, NOTUSED, NOTUSED, NOTUSED);
+        return MAKE_SYSCALL(IOCTL_SYSCALL, "syscall|ioctl", (uint64_t) fd, (uint64_t) request, (uint64_t) TRANSLATE_GUEST_POINTER_TO_HOST(arg), NOTUSED, NOTUSED, NOTUSED);
```

Comment:
ioctl is extremely weird and uses var args, i dont think this is handling that correclty

Reply:
Ok so from discussion we just want to handle FIONBIO and FIOASYNC which both need pointer translation. 

lets assert and fail for other cases so we know they come up